docs/src/modules/components/Link.tsx

```diff
@@ -64,7 +64,7 @@ export type LinkProps = {
   linkAs?: NextLinkProps['as']; // Useful when the as prop is shallow by styled().
   noLinkStyle?: boolean;
 } & Omit<NextLinkComposedProps, 'to' | 'linkAs' | 'href'> &
-  Omit<MuiLinkProps, 'href'>;
+  Omit<MuiLinkProps, 'href' | 'component'>;
```

Comment:
Why do we need to omit `component` here? Is it overlapping with some other `component` prop where MUI's `LinkProps` is used?

Reply:
fixed here https://github.com/mui/material-ui/pull/35924/commits/84ecfff03d5832b911055582004eabb8714a39b3

Reply:
I'm unable to give you exact reason but as far as i remember (i worked on this long long ago), `component` is omitted here because there was some overlapping of `component` type where it's being used just as you said.

Reply:
What if you undo this change? Is there a type error somewhere? Maybe you could try it.

Reply:
yes there is typescript error on undoing this change. I can't paste the typescript error here as it's really big but if you want to see error by yourself, undo the change and open /docs/pages/blog.tsx file.